orchestrator/domain/base.py

```diff
@@ -416,6 +422,47 @@ def _save_instances(
 
         return saved_instances, depends_on_instances
 
+    @classmethod
+    def get_properties(cls) -> list[Any]:
+        def is_serializable_property(prop: Any) -> bool:
+            return isinstance(cls.__dict__[prop], property)
+
+        return [prop for prop in cls.__dict__ if is_serializable_property(prop)]
+
+    def dict(
+        self,
+        *,
+        include: Any | None = None,
+        exclude: Any | None = None,
+        by_alias: bool = False,
+        skip_defaults: bool | None = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
```

Comment:
Is it possible to use kwargs instead?

```python
def dict(self, *, **kwargs) -> dict[str, Any]:
    attribs = super().dict(**kwargs)
    
    if include := kwargs.get('include'):
        ...
```

Reply:
Probably. Was just following the suggestion from one of the comments in a Pydantic issue. I'm fine with both options.

Reply:
Changed